### PR TITLE
[HIP] enable global lock arrays

### PR DESCRIPTION
**Pull request by Rombur**
This PR enables:
 - the global lock arrays
 - all the incremental tests
 - more tests of the testsuite. Some are modified to reduce the size of the loop.

As far as I can tell the tests that still do not pass are tests that either use `complex` or tests which pass for smaller sizes.

### CODE REVIEW

File: core/unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -326,8 +326,6 @@ if(Kokkos_ENABLE_HIP)
   # FIXME_HIP
   LIST(REMOVE_ITEM HIP_SOURCES
     hip/TestHIP_AtomicOperations_complexdouble.cpp
```
Comment (masterleinad) [inline core/unit_test/CMakeLists.txt:328]: What's the problem with this test?

File: core/unit_test/TestTeamTeamSize.hpp [hunk 2]
```diff
@@ -121,12 +121,21 @@ void test_team_policy_max_recommended_static_size(int scratch_size) {
                            .set_scratch_size(0, Kokkos::PerTeam(scratch_size)),
                        FunctorFor<T, N, PolicyType, S>());
   MyArray<T, N> val;
+  double n_leagues = 10000;
+  // FIXME_HIP
```
Comment (masterleinad) [inline core/unit_test/TestTeamTeamSize.hpp:125]: Any idea what the issue here is?

File: core/unit_test/TestTeamVectorRange.hpp [hunk 3]
```diff
@@ -477,8 +477,11 @@ bool Test(int test) {
            test_scalar<long long int, ExecutionSpace>(317, team_size, test);
   passed = passed && test_scalar<float, ExecutionSpace>(317, team_size, test);
   passed = passed && test_scalar<double, ExecutionSpace>(317, team_size, test);
+  // FIXME_HIP
```
Comment (masterleinad) [inline core/unit_test/TestTeamVectorRange.hpp:480]: ... and here?

**Review by masterleinad — COMMENTED** [review]
It would be good if you could provide comments on why these tests still fail in case you know.

File: core/unit_test/CMakeLists.txt (hunk 1 above)
Comment (Rombur) [reply core/unit_test/CMakeLists.txt:328]: I haven't really investigate the problem with `complex` numbers yet.

File: core/unit_test/TestTeamTeamSize.hpp (hunk 2 above)
Comment (Rombur) [reply core/unit_test/TestTeamTeamSize.hpp:125]: That's the segfault that we reported to AMD and we talked about during the last group meeting

**Review by crtrott — APPROVED** [review]
Can you open an explicit issue to track the complex number issue? Otherwise is ok.

**Comment by Rombur** [comment]
I have open an issue

**Comment by Rombur** [comment]
Retest this please

**Pull request merged**